cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,259 @@
+            kwargs.update(attributes)
+        super(CloudEvent, self).__init__(data=data, **kwargs)
+
+    class Config:
+        extra: str = "allow"  # this is the way we implement extensions
+        schema_extra = {
+            "examples": [
+                {
+                    "specversion": "1.0",
+                    "type": "com.github.pull_request.opened",
+                    "source": "https://github.com/cloudevents/spec/pull",
+                    "subject": "123",
+                    "id": "A234-1234-1234",
+                    "time": "2018-04-05T17:31:00Z",
+                    "comexampleextension1": "value",
+                    "comexampleothervalue": 5,
+                    "datacontenttype": "text/xml",
+                    "data": '<much wow="xml"/>',
+                },
+                {
+                    "specversion": "1.0",
+                    "id": "O234-345-890",
+                    "source": "https://dummyteashop.com/1234/order",
+                    "type": "com.dummyteashop.tea.Order",
+                    "subject": "green-tea",
+                    "time": "2020-11-25T09:05:00Z",
+                    "datacontenttype": "application/json",
+                    "data": {
+                        "tea_type": "green-tea",
+                        "kettle": {
+                            "temperature": "100c",
+                            "size": "large",
+                        },
+                        "cup": {
+                            "color": "red",
+                            "size": "medium",
+                        },
+                    },
+                },
+            ]
+        }
+        json_dumps = _ce_json_dumps
+        json_loads = _ce_json_loads
+
+    def _get_attributes(self) -> typing.Dict[str, typing.Any]:
+        return {
+            key: conversion.encode_attribute_value(value)
+            for key, value in self.__dict__.items()
+            if key != "data"
+        }
+
+    def _get_data(self) -> typing.Optional[typing.Any]:
+        return self.data
+
+    def __setitem__(self, key: str, value: typing.Any) -> None:
+        """
+        Set event attribute value
+
+        MUST NOT set event data with this method, use `.data` member instead
+
+        Method SHOULD mimic `cloudevents.http.event.CloudEvent` interface
+
+        :param key: Event attribute name
+        :param value: New event attribute value
+        """
+        if key != "data":  # to mirror the behaviour of the http event
+            setattr(self, key, value)
+
+    def __delitem__(self, key: str) -> None:
+        """
+        SHOULD raise `KeyError` if no event attribute for the given key exists.
+
+        Method SHOULD mimic `cloudevents.http.event.CloudEvent` interface
+        :param key:  The event attribute name.
+        """
+        if key == "data":
+            raise KeyError(key)  # to mirror the behaviour of the http event
```

Comment:
Why do we raise an error here, but we don't for `setitem`? This is not consistent.

Reply:
@sasha-tkachev what about this one?

Reply:
We're keeping the implementation consistent.